### ISSUE TITLE
Infinite loop while fetching trades from Bitstamp

### ISSUE THREAD

**Issue by realart** [askmike/gekko#228]
Left the bot running overnight and in the morning I found it spewing out loads of DEBUG entries at high speed, looping around the same timeframe, not getting anywhere:

```
2014-03-09 14:21:51 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    done with this batch (1)
2014-03-09 14:21:51 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:51 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:51 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:51 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:51 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    done with this batch (1)
2014-03-09 14:21:51 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:51 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    done with this batch (1)
2014-03-09 14:21:51 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:51 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:51 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:51 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    done with this batch (1)
2014-03-09 14:21:51 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:51 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    done with this batch (1)
2014-03-09 14:21:51 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:51 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    done with this batch (1)
2014-03-09 14:21:51 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:51 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    done with this batch (1)
2014-03-09 14:21:51 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:51 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    done with this batch (1)
2014-03-09 14:21:51 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:51 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:51 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:51 (DEBUG):    done with this batch (1)
2014-03-09 14:21:51 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:52 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:52 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:52 (DEBUG):    done with this batch (1)
2014-03-09 14:21:52 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:52 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:52 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:52 (DEBUG):    done with this batch (1)
2014-03-09 14:21:52 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:52 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:52 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:52 (DEBUG):    done with this batch (1)
2014-03-09 14:21:52 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:52 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:52 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:52 (DEBUG):    done with this batch (1)
2014-03-09 14:21:52 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:53 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:53 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:53 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:53 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:53 (DEBUG):    done with this batch (1)
2014-03-09 14:21:53 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:53 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:53 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:53 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:53 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:53 (DEBUG):    done with this batch (1)
2014-03-09 14:21:53 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:53 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:53 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:54 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:54 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:54 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:54 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:55 (DEBUG):    Bitstamp returned an error, retrying..
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Bitstamp returned an error, retrying..
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:55 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:55 (DEBUG):    done with this batch (1)
2014-03-09 14:21:55 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:56 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:56 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:56 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:56 (DEBUG):    done with this batch (1)
2014-03-09 14:21:56 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:56 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:56 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:56 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:56 (DEBUG):    done with this batch (1)
2014-03-09 14:21:56 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:56 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:56 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:57 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:57 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:57 (DEBUG):    done with this batch (1)
2014-03-09 14:21:57 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:57 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:57 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:57 (DEBUG):    done with this batch (1)
2014-03-09 14:21:57 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:57 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:57 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:57 (DEBUG):    done with this batch (1)
2014-03-09 14:21:57 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:21:58 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:59 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:59 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
2014-03-09 14:21:59 (DEBUG):    Fetched 122 new trades, from 2014-03-09 11:21:26
 UTC to 2014-03-09 12:20:18 UTC
2014-03-09 14:21:59 (DEBUG):    minimum trade treshold: 2014-03-09 12:20:18 UTC
2014-03-09 14:21:59 (DEBUG):    done with this batch (1)
2014-03-09 14:21:59 (DEBUG):    Processed trades, sleeping for a minute...
2014-03-09 14:22:00 (DEBUG):    Requested BTC/USD trade data from Bitstamp ...
```

**Comment by BabyDino** [comment]
I have the same issue with the default config..

**Comment by askmike** [comment]
fixed in 0.2!
